tsdb/head.go

```diff
@@ -1419,11 +1465,14 @@ func (a *headAppender) Append(ref uint64, lset labels.Labels, t int64, v float64
 // AppendExemplar for headAppender assumes the series ref already exists, and so it doesn't
 // use getOrCreate or make any of the lset sanity checks that Append does.
 func (a *headAppender) AppendExemplar(ref uint64, _ labels.Labels, e exemplar.Exemplar) (uint64, error) {
+	switch a.head.exemplars.(type) {
+	case noopExemplarStorage:
+		return 0, nil
+	}
```

Comment:
Do we need this if we are checking the options anyway?

Reply:
nope